tests/unit/parallel/datapar_algorithms/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2016 Hartmut Kaiser
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+set(tests)
+
+if(HPX_WITH_VC_DATAPAR)
+  set(tests
+      count_datapar
+      countif_datapar
+      foreach_datapar
+      foreachn_datapar
+      inner_product_datapar
+      transform_datapar
+      transform_binary_datapar
+      transform_binary2_datapar
+     )
+endif()
+
+foreach(test ${tests})
+  set(sources
+      ${test}.cpp)
+
+  set(${test}_PARAMETERS THREADS_PER_LOCALITY 4)
+
+  source_group("Source Files" FILES ${sources})
+
+  # add example executable
+  add_hpx_executable(${test}_test
+                     SOURCES ${sources}
+                     ${${test}_FLAGS}
+                     EXCLUDE_FROM_ALL
+                     HPX_PREFIX ${HPX_BUILD_PREFIX}
+                     FOLDER "Tests/Unit/Parallel/DataParAlgorithms")
+
+  if(NOT MSVC)
+    foreach(_flag ${Vc_COMPILE_FLAGS})
+      set_target_properties(${test}_test_exe PROPERTIES COMPILE_FLAGS ${_flag})
+    endforeach()
+  endif()
+
+  foreach(_flag ${Vc_ARCHITECTURE_FLAGS})
+    set_target_properties(${test}_test_exe PROPERTIES COMPILE_FLAGS ${_flag})
+  endforeach()
```

Comment:
Same as for the inner_product benchmark. I think it would be wise to have these properties set globally. Otherwise this will be a subtile source of errors.